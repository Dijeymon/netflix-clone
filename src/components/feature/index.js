/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prop-types */
import React from 'react';
import { Container, Title, Subtitle } from './styles/feature';

export default function Features({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Features.Title = function FeaturesTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Features.Subtitle = function FeaturesSubtitle({ children, ...restProps }) {
  return <Subtitle {...restProps}>{children}</Subtitle>;
};
