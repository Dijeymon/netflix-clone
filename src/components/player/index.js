/* eslint-disable no-shadow */
/* eslint-disable jsx-a11y/media-has-caption */
/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-props-no-spreading */
import React, { createContext, useContext, useState } from 'react';
import ReactDOM from 'react-dom';
import { Container, Button, Overlay, Inner, Close } from './styles/player';

export const PlayerContext = createContext();

export default function Player({ children, ...restProps }) {
  const [showPlayer, setShowPlayer] = useState(false);

  return (
    <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
      <Container {...restProps}>{children}</Container>
    </PlayerContext.Provider>
  );
}

Player.Video = function PlayerVideo({ src }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return showPlayer
    ? ReactDOM.createPortal(
        <Overlay onClick={() => setShowPlayer(false)} data-testid="player">
          <Inner>
            <video id="netflix-player" controls>
              <source src={src} type="video/mp4" />
            </video>
            <Close />
          </Inner>
        </Overlay>,
        document.body
      )
    : null;
};

Player.Button = function Player({ ...restProps }) {
  // eslint-disable-next-line no-unused-vars
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return (
    <Button
      {...restProps}
      onClick={() => setShowPlayer((showPlayer) => !showPlayer)}
    >
      Play
    </Button>
  );
};
