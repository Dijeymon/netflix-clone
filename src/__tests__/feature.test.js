import React from 'react';
import { render } from '@testing-library/react';
import { Features } from '../components';

describe('<Features />', () => {
  it('renders the <Features /> with populated data', () => {
    const { container, getByText } = render(
      <Features>
        <Features.Title>
          Unlimited films, TV programmes and more.
        </Features.Title>
        <Features.Subtitle>
          Watch anywhere. Cancel at any time.
        </Features.Subtitle>
      </Features>
    );

    expect(getByText('Unlimited films, TV programmes and more.')).toBeTruthy();
    expect(getByText('Watch anywhere. Cancel at any time.')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Features /> with just a title', () => {
    const { container, getByText, queryByText } = render(
      <Features>
        <Features.Title>
          Unlimited films, TV programmes and more.
        </Features.Title>
      </Features>
    );

    expect(getByText('Unlimited films, TV programmes and more.')).toBeTruthy();
    expect(queryByText('Watch anywhere. Cancel at any time.')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
