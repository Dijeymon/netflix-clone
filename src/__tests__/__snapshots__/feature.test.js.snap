// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Features /> renders the <Features /> with just a title 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 caFRqP"
>
  <h1
    class="feature__Title-sc-6swrp6-1 kfqHuI"
  >
    Unlimited films, TV programmes and more.
  </h1>
</div>
`;

exports[`<Features /> renders the <Features /> with populated data 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 caFRqP"
>
  <h1
    class="feature__Title-sc-6swrp6-1 kfqHuI"
  >
    Unlimited films, TV programmes and more.
  </h1>
  <h2
    class="feature__Subtitle-sc-6swrp6-2 bjYOhA"
  >
    Watch anywhere. Cancel at any time.
  </h2>
</div>
`;
